sound/soc/sof/topology.c

```diff
@@ -3046,8 +3046,25 @@ static void sof_complete(struct snd_soc_component *scomp)
 static int sof_manifest(struct snd_soc_component *scomp, int index,
 			struct snd_soc_tplg_manifest *man)
 {
-	/* not currently parsed */
-	return 0;
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
+
+	/* backward compltible with tplg without ABI info */
+	if (man->priv.size == 0)
+		return 0;
+
+	if (man->priv.size == 3) {
+		dev_info(sdev->dev,
+			 "Topology: ABI %d:%d:%d Kernel ABI %d:%d:%d\n",
+			 man->priv.data[0], man->priv.data[1],
+			 man->priv.data[2], SOF_ABI_MAJOR, SOF_ABI_MINOR,
+			 SOF_ABI_PATCH);
+		if (man->priv.data[0] == SOF_ABI_MAJOR)
+			return 0;
```

Comment:
what happens if the kernel sees a topology file with a more recent ABI_MINOR value? There will be tokens and values it cannot parse, should we reject such cases?

Reply:
@plbossart Should it go to ABI_MAJOR updated if it is not backward compatible? Sorry, I am not very clear with when should we update the ABI major, minor, or patch number.

Reply:
@bardliao for some new tokens, we update only the minor version when we use the reserved fields to add new fields. It is ok to compare just the major version for compatibility between the kernel and fw in this case as the struct size itself wouldnt change. But like Pierre mentioned, if the field was added recently there probably wouldnt be a corresponding token in topology, so all we would get is 0's for the field. So we should check for both major and minor version when checking for compatibility between tplg and kernel

Reply:
Thanks @ranj063 @plbossart the PR has updated.